.github/workflows/build-and-test.yml

```diff
@@ -593,3 +593,113 @@ jobs:
         if: ${{ failure() && runner.debug }}
         uses: canonical/action-tmate@main
 
+  juju-storage-cluster-test:
+    needs:
+      - lint
+      - unit-test
+      - build
+    name: Juju multi node storage test 
+    runs-on: [self-hosted, xlarge]
+    steps:
+      - name: Download charm
+        uses: actions/download-artifact@v3
+        with:
+          name: charms
+          path: ~/artifacts/
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup LXD
+        uses: canonical/setup-lxd@v0.1.1
+        with:
+          # pin lxd to LTS release.
+          channel: 5.21/stable
+
+      - name: Install dependencies
+        run: ./tests/scripts/ci_helpers.sh install_deps
+
+      - name: Install MicroCeph charm
+        run: |
+          date
+          mv ~/artifacts/microceph.charm ./microceph.charm
+          juju deploy ./tests/bundles/multi_node_juju_storage.yaml 
+          # wait for charm to bootstrap and OSD devices to enroll.
+          ./tests/scripts/ci_helpers.sh wait_for_microceph_bootstrap
+          date
+
+      - name: Show Juju status
+        run: |
+          set -eux
+          juju status
+
+      - name: Remove Unit microceph/2
+        run: |
+          set -x
+          disk_list=$(juju ssh microceph/leader -- "sudo microceph disk list --json")
+          osd_count=$(echo $disk_list | jq '.ConfiguredDisks[].location' | grep -c "\-2")
```

Comment:
Very minor nit: suggest to grep for a longer string then "-2" for clarity

Reply:
ack